### PR TITLE
Remove swizzle node for 1.39

### PR DESCRIPTION
**Pull request by jstone-lucasfilm**
This changelist removes the swizzle node from the MaterialX data libraries for v1.39, adding an upgrade pathway for instances of this node in legacy documents.

Additionally, it addresses a minor inconsistency in the specification for the extract and extractrowvector nodes, which take zero-based indices rather than positive-valued selections.

### CODE REVIEW

**Comment by jstone-lucasfilm** [comment]
Attaching updated comparison renders in GLSL and OSL:

[MaterialXRenderTests_04_28_2024.pdf](https://github.com/AcademySoftwareFoundation/MaterialX/files/15143611/MaterialXRenderTests_04_28_2024.pdf)

File: source/MaterialXCore/Version.cpp [hunk 1]
```diff
@@ -1077,6 +1097,107 @@ void Document::upgradeVersion()
                     }
                 }
             }
+            else if (nodeCategory == "swizzle")
+            {
+                InputPtr inInput = node->getInput("in");
+                InputPtr channelsInput = node->getInput("channels");
+                if (inInput &&
```
Comment (ld-kerley) [inline source/MaterialXCore/Version.cpp:1104]: Do you need to handle the case where `inInput` doesn't exist?  All the node defs appear to return `0`, but it's possible the `swizzle` node was wired in to a port that doesn't have `0` as its default?

**Comment by jstone-lucasfilm** [comment]
@ld-kerley Absolutely, and there's a dedicated file in the MDL shader generation system that can now be removed as well, but I'm taking this change in stages!

**Review by ld-kerley — COMMENTED** [review]
The upgrade function is really cleanly written :) - I was really worried it would be a lot more complex than that. Nice work!

File: source/MaterialXCore/Version.cpp (hunk 1 above)
Comment (jstone-lucasfilm) [reply source/MaterialXCore/Version.cpp:1104]: That's a good point, and perhaps we should handle this case by converting the `swizzle` node to a `constant` node containing the appropriate values.  I haven't encountered this case in our test assets, but it's theoretically possible.

**Pull request merged**